File: market-backend/src/main/java/com/ais/marketbackend/dashboard/application/services/interfaces/DashboardService.java
package com.ais.marketbackend.dashboard.application.services.interfaces;

import com.ais.marketbackend.dashboard.application.dtos.DashboardGrupoResumen;
import com.ais.marketbackend.dashboard.application.dtos.DashboardResumen;

public interface DashboardService {

    DashboardResumen obtenerResumen(Long tiendaId);

    /** Agrega {@link #obtenerResumen(Long)} de cada tienda del grupo. Exige acceso al grupo. */
    DashboardGrupoResumen obtenerResumenGrupo(Long grupoId);
}

File: market-backoffice/src/services/dashboard.service.ts
import { apiClient } from '@/services/http/ApiClient'
import { API_ENDPOINTS } from '@/config/endpoints'
import type { DashboardGrupoResumen, DashboardResumen } from '@/types/dashboard'

class DashboardService {
  obtenerResumen(tiendaId: number) {
    return apiClient.get<DashboardResumen>(API_ENDPOINTS.dashboard.porTienda(tiendaId))
  }

  obtenerResumenGrupo(grupoId: number) {
    return apiClient.get<DashboardGrupoResumen>(API_ENDPOINTS.dashboard.porGrupo(grupoId))
  }
}

export const dashboardService = new DashboardService()
